eegstats/R/main.R

```diff
@@ -1,69 +1,289 @@
+
+# 5. grand average ersp
+#' grand_average_ersp
+#' @description  The function \emph{grand_average_ersp} computes the grand average  ERSP signal for all conditions and all frequencies from one electrode.
+#' @param my_erspdata Input Matrix with shape: subject ID x condition x timex frequency (from only one electrode).
+#' @details For more details, see Wikipedie page: \href{https://sccn.ucsd.edu/wiki/Chapter_11:_Time/Frequency_decomposition}{article}
+#' @return A matrix that contains the grand average ERSP for all conditions and all frequencies of one electrode. The matrix will be of shape condition x time x frequency.
+#' @export grand_average_ersp_data
+#' @seealso Add example of similar functions [still to be done].
+
+grand_average_ersp <- function(my_erspdata) {
+
+  cond <- unique(my_erpdata[,2])
+  time <- unique(my_erpdata[,3])
+
+  grand_average_ersp_data <- array(NA, dim = c(length(conditions),length(timepoints), 20))
+  for (c in 1:length(cond)) { # all conditions
+
+    for (t in 1:length(time)) { # all timepoints
+
+      grand_average_ersp_data[c, t,] <-  colMeans(my_erpdata[my_erpdata[,2] == cond[c] & my_erpdata[,3] == time[t], c(4:ncol(my_erpdata))])
+    }
+  }
+  return(grand_average_ersp_data)
 }
+
+grand_average_ersp_data <- grand_average_ersp(my_erpdata)
+
+
+
+# 6. create time frequency plot
+#' grand_average_ersp_plot
+#' @description  The function \emph{grand_average_ersp_plot} plots the grand average ERSP as a time frequency plot.
+#' @param my_erspdata Input Matrix with shape: subject ID x condition x timex frequency (from only one electrode).
+#' @param grand_average_ersp_data A matrix that contains the grand average ERSP for all conditions and all frequencies of one electrode. The matrix will be of shape condition x time x frequency.
+#' @param ncond A numeric value that specifies which condition should be plotted (e.g. 1 for condition 1)
+#' @return A time - frequency plot with the grand average ERSP signal.
+#' @seealso Add example of similar functions [still to be done].
+
+# I wanted to include an xample output figure in the documentation:
+#\if{html}{\figure{erp.png}{options: width=100 alt="erp.png"}}
+#\if{latex}{\figure{erp.png}{options: width=7cm}}
+# This didn't work - any suggestions how to add images to R function documentation?
+
+grand_average_ersp_plot <- function(my_erspdata, grand_average_ersp_data, n_cond) {
+
+  cond <- unique(my_erpdata[,2])
+  time <- unique(my_erpdata[,3])
+
+  # get the right label for the heatmap
+  labRow_label = seq(from = time[1], to = time[length(time)], by = 1)
+
+  # add correct timepoints (as in original data)
+  desired_timepoints <- seq(from = time[1], to = time[length(time)], by = 400,)
+
+  index_time <- c()
+
+  for (i in 1:length(desired_timepoints)) {
+
+    j <- desired_timepoints[i]
+
+    index_time[i] <- which(time == j)
+
+  }
+
+  labRow_label <- NA
+  labRow_label[index_time] <- desired_timepoints
```

Comment:
The teachers stressed that we should stay consistent in our name giving and labeling. 
Suddenly using camelCase instead of your usual_choice could cost you style points